core/src/main/java/com/linecorp/armeria/server/AbstractServiceBindingBuilder.java

```diff
@@ -89,197 +62,117 @@
+        routeBindingBuilder.head(pathPattern);
         return this;
     }
 
-    /**
-     * Sets the path pattern and {@link HttpMethod#TRACE} that a {@link Service} will be bound to and support.
-     *
-     * @throws IllegalArgumentException if the specified path pattern is invalid
-     */
+    @Override
     public AbstractServiceBindingBuilder trace(String pathPattern) {
-        addRouteBuilder(pathPattern, EnumSet.of(TRACE));
+        routeBindingBuilder.trace(pathPattern);
         return this;
     }
 
-    /**
-     * Sets the path pattern and {@link HttpMethod#CONNECT} that a {@link Service} will be bound to and support.
-     *
-     * @throws IllegalArgumentException if the specified path pattern is invalid
-     */
+    @Override
     public AbstractServiceBindingBuilder connect(String pathPattern) {
-        addRouteBuilder(pathPattern, EnumSet.of(CONNECT));
+        routeBindingBuilder.connect(pathPattern);
         return this;
     }
 
-    /**
-     * Sets {@link MediaType}s that a {@link Service} will consume. If not set, the {@link Service}
-     * will accept all media types.
-     */
+    @Override
     public AbstractServiceBindingBuilder consumes(MediaType... consumeTypes) {
-        consumes(ImmutableSet.copyOf(requireNonNull(consumeTypes, "consumeTypes")));
+        routeBindingBuilder.consumes(consumeTypes);
         return this;
     }
 
-    /**
-     * Sets {@link MediaType}s that a {@link Service} will consume. If not set, the {@link Service}
-     * will accept all media types.
-     */
+    @Override
     public AbstractServiceBindingBuilder consumes(Iterable<MediaType> consumeTypes) {
-        ensureUniqueMediaTypes(consumeTypes, "consumeTypes");
-        this.consumeTypes = ImmutableSet.copyOf(consumeTypes);
+        routeBindingBuilder.consumes(consumeTypes);
         return this;
     }
 
-    /**
-     * Sets {@link MediaType}s that a {@link Service} will produce to be used in
-     * content negotiation. See <a href="https://tools.ietf.org/html/rfc7231#section-5.3.2">Accept header</a>
-     * for more information.
-     */
+    @Override
     public AbstractServiceBindingBuilder produces(MediaType... produceTypes) {
-        produces(ImmutableSet.copyOf(requireNonNull(produceTypes, "produceTypes")));
+        routeBindingBuilder.produces(produceTypes);
         return this;
     }
 
-    /**
-     * Sets {@link MediaType}s that a {@link Service} will produce to be used in
-     * content negotiation. See <a href="https://tools.ietf.org/html/rfc7231#section-5.3.2">Accept header</a>
-     * for more information.
-     */
+    @Override
     public AbstractServiceBindingBuilder produces(Iterable<MediaType> produceTypes) {
-        ensureUniqueMediaTypes(produceTypes, "produceTypes");
-        this.produceTypes = ImmutableSet.copyOf(produceTypes);
+        routeBindingBuilder.produces(produceTypes);
         return this;
     }
 
+    @Override
+    public List<Route> buildRoute() {
```

Comment:
How about renaming to `toRouteList()`, because otherwise users will see  more than one `build*()` methods in the subclasses? Alternatively, we could turn `RouteBindingBuilder` an abstract class and make `buildRoute()` package-private.

Reply:
I prefer renaming `RouteBindingBuilder` to `AbstractBindingBuilder` and making `buildRoute()` package-private. (And please rename `buildRoute()` to `buildRouteList()` :wink:)

Reply:
I like `buildRouteList()` 😄 Thanks for a nice suggestion.